### PR TITLE
TA- 556-accessibility tests idam web public

### PR DESCRIPTION
**Pull request by radhikahollakv**
**Before creating a pull request make sure that:**

### Change description ###

This PR includes tests for accessibility for Account Lock and User login scenario 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -152,7 +152,7 @@ allprojects {
     description = 'Executes non-destructive smoke tests against a running instance'
   }
 
-  task functional(dependsOn: [':codeceptFunctional', ':pa11y']) {
+  task functional(dependsOn: [':codeceptFunctional', ':pa11y', 'functional1']) {
```
Comment (shravanmechineni) [inline build.gradle:155]: There is no functional1 task in the gradle build? why is it required?

File: package.json [hunk 2]
```diff
@@ -11,11 +11,12 @@
     "codeceptjs": "^3.0.4",
     "codeceptjs-resemblehelper": "^1.9.3",
     "deep-equal-in-any-order": "^1.0.28",
+    "html_codesniffer": "^2.5.1",
     "jwt-decode": "^3.1.2",
     "mocha-junit-reporter": "^2.0.0",
     "mocha-multi": "^1.1.3",
     "mochawesome": "^6.2.1",
-    "nightmare": "^3.0.2",
+    "nightmare": "^2.8.0",
```
Comment (shravanmechineni) [inline package.json:19]: Is this version change intended and required for accessibility tests?

File: Jenkinsfile_nightly [hunk 3]
```diff
@@ -40,12 +40,20 @@ withNightlyPipeline(type, product, component) {
     env.TEST_URL = params.URL_TO_TEST
     env.IDAMAPI = params.API_URL_TO_TEST
     loadVaultSecrets(secrets)
-    enableSecurityScan()
+    //enableSecurityScan()
```
Comment (shravanmechineni) [inline Jenkinsfile_nightly:43]: Why is it commented?

File: Jenkinsfile_nightly [hunk 4]
```diff
@@ -40,12 +40,20 @@ withNightlyPipeline(type, product, component) {
...
     enableMutationTest()
     enableFullFunctionalTest(200)
-    enableCrossBrowserTest()
+    //enableCrossBrowserTest()
```
Comment (shravanmechineni) [inline Jenkinsfile_nightly:46]: same here?

File: src/test/js/reporters/accessibility-reporter/Report.html [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
+<!--[if gt IE 8]><!--> <html class="no-js"> <!--<![endif]-->
+<head>
+  <meta charset="utf-8">
+  <!-- <meta http-equiv="Content-Security-Policy"
+      content="default-src *; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' 'unsafe-eval'
+       https://cdnjs.cloudflare.com https://ajax.googleapis.com http://www.google.com"> -->
+  <script src="resources/angular.min.js"></script>
+
+  <title>EXUI Web Accesibility Test Report</title>
```
Comment (shravanmechineni) [inline src/test/js/reporters/accessibility-reporter/Report.html:13]: Can we change to IDAM from EXUI?

File: build.gradle (hunk 1 above)
Comment (radhikahollakv) [reply build.gradle]: This was something missed to remove This has been removed now in the latest push

File: package.json (hunk 2 above)
Comment (radhikahollakv) [reply package.json]: I had not made this change not sure how it got changed. However, I have updated nightmare back to ^3.0.2

File: src/test/js/reporters/accessibility-reporter/Report.html (hunk 5 above)
Comment (radhikahollakv) [reply src/test/js/reporters/accessibility-reporter/Report.html]: This has been updated in the latest push

File: Jenkinsfile_nightly (hunk 4 above)
Comment (radhikahollakv) [reply Jenkinsfile_nightly]: this has been now uncommented.

File: Jenkinsfile_nightly (hunk 3 above)
Comment (radhikahollakv) [reply Jenkinsfile_nightly]: Uncommented

File: src/test/js/shared/accessibility/runner.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+const fs = require('fs');
+const testConfig = require('../../config/test_data.js');
+
+const result = {
+  PASSED: 'passed',
+  FAILED: 'failed',
+};
+
+const resultObj = {
+  appName: 'FPL',
```
Comment (shravanmechineni) [inline src/test/js/shared/accessibility/runner.js:11]: Need to be changed to **idam-web-public**

File: src/test/js/reporters/accessibility-reporter/Report.html [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+<body>
+<div ng-app="myApp" ng-controller="myCtrl">
+  <div class="header">
+    <div class="titleText">
+      <div>
+        <i class="fas fa-universal-access" style="font-size:60px;color:white;"> </i>
+      </div>
+
+      <div>
+        <div id="reportHeaderTitle">Accessibility Test Report for Expert UI</div>
```
Comment (shravanmechineni) [inline src/test/js/reporters/accessibility-reporter/Report.html:135]: Expert UI to be changed to **IDAM Web Public UI**.

File: src/test/js/shared/idam_helper.js [hunk 8]
```diff
@@ -701,6 +703,16 @@ class IdamHelper extends Helper {
             console.log(err);
         });
     }
+    async runAccessibilityTest() {
+        if (!testConfig.TestForAccessibility) {
+            return;
+        }
+        const url = await this.helpers[Helper].grabCurrentUrl();
```
Comment (shravanmechineni) [inline src/test/js/shared/idam_helper.js:710]: Helper should be mentioned explicitly i.e. Puppeteer.

File: src/test/js/shared/idam_helper.js [hunk 9]
```diff
@@ -701,6 +703,16 @@ class IdamHelper extends Helper {
...
+        const {page} = await this.helpers[Helper];
```
Comment (shravanmechineni) [inline src/test/js/shared/idam_helper.js:711]: Same here.

File: src/test/js/reporters/accessibility-reporter/customReporter.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+  fs.copyFileSync(__dirname + '/resources/angular.min.js', resourceDir + 'angular.min.js');
+  fs.copyFileSync(__dirname + '/resources/css/all.css', cssDir + 'all.css');
+  fs.copyFileSync(__dirname + '/resources/webfonts/fa-solid-900.woff2', webfontsDir + 'fa-solid-900.woff2');
+}
+
+function consoleReport(reportjson) {
+  /* eslint-disable no-console */
+  console.log('\t Total tests : ' + reportjson.tests.length);
+  console.log('\t Passed tests : ' + reportjson.passCount);
+  console.log('\t Failed tests : ' + reportjson.passCount);
```
Comment (shravanmechineni) [inline src/test/js/reporters/accessibility-reporter/customReporter.js:56]: reportjson.failCount?

File: src/test/js/shared/accessibility/runner.js (hunk 6 above)
Comment (radhikahollakv) [reply src/test/js/shared/accessibility/runner.js]: This has been updated now. Thank you!

File: src/test/js/reporters/accessibility-reporter/Report.html (hunk 7 above)
Comment (radhikahollakv) [reply src/test/js/reporters/accessibility-reporter/Report.html]: This has been updated now. Thank you!

File: src/test/js/shared/idam_helper.js (hunk 8 above)
Comment (radhikahollakv) [reply src/test/js/shared/idam_helper.js]: This has been updated now. Thank you!

File: src/test/js/shared/idam_helper.js (hunk 9 above)
Comment (radhikahollakv) [reply src/test/js/shared/idam_helper.js]: This has been updated now. Thank you!

File: src/test/js/reporters/accessibility-reporter/customReporter.js (hunk 10 above)
Comment (radhikahollakv) [reply src/test/js/reporters/accessibility-reporter/customReporter.js]: This has been updated now. Thank you!

File: src/test/js/ejudiciary_login_test.js [hunk 11]
```diff
@@ -24,12 +24,14 @@ Scenario('@functional @ejudiciary As an ejudiciary user, I can login into idam t
...
     I.waitForText('Sign in');
+    await I.runAccessibilityTest();
     I.fillField('loginfmt', TestData.EJUDICIARY_TEST_USER_USERNAME);
     I.click('Next');
     I.waitForText('Enter password');
     I.fillField('passwd', TestData.EJUDICIARY_TEST_USER_PASSWORD);
     I.click('Sign in');
-    I.waitForText('Stay signed in?');
+    I.waitForText('Stay signed in?')
+    await I.runAccessibilityTest();
```
Comment (shravanmechineni) [inline src/test/js/ejudiciary_login_test.js:34]: it's not an idam page?

File: src/test/js/ejudiciary_login_test.js [hunk 12]
```diff
@@ -67,9 +69,11 @@ Scenario('@functional @ejudiciary As an ejudiciary user, I should be able to log
     I.waitInUrl('/oauth2/authorize');
     I.waitForText('Sign in');
     I.fillField('loginfmt', TestData.EJUDICIARY_TEST_USER_USERNAME);
+    await I.runAccessibilityTest();
```
Comment (shravanmechineni) [inline src/test/js/ejudiciary_login_test.js:72]: same.

File: src/test/js/self_registration_test.js [hunk 13]
```diff
@@ -78,16 +83,19 @@ Scenario('@functional @selfregister @welshLanguage Account already created (no l
     I.amOnPage(selfRegUrl);
     I.waitInUrl('users/selfRegister');
     I.waitForText('Create an account or sign in');
-
+    await I.runAccessibilityTest();
```
Comment (shravanmechineni) [inline src/test/js/self_registration_test.js:86]: already covered.

Comment (shravanmechineni) [inline src/test/js/self_registration_test.js:86]: already covered above.

File: src/test/js/self_registration_test.js [hunk 14]
```diff
@@ -78,16 +83,19 @@ Scenario('@functional @selfregister @welshLanguage Account already created (no l
...
     I.fillField('lastName', randomUserLastName);
     I.fillField('email', citizenEmail);
+    await I.runAccessibilityTest();
     I.click("Continue");
 
     I.waitForText('Check your email');
+    await I.runAccessibilityTest();
     const emailResponse = await I.getEmailFromNotify(citizenEmail);
     assert.equal('You already have an account', emailResponse.subject);
+    await I.runAccessibilityTest();
```
Comment (shravanmechineni) [inline src/test/js/self_registration_test.js:98]: not a page, redundant.

**Review by kremi — APPROVED** [review]

**Pull request closed without merging**